examples/common/chip-app-server/RendezvousServer.cpp

```diff
@@ -115,4 +122,25 @@ void RendezvousServer::OnRendezvousMessageReceived(PacketBuffer * buffer)
     chip::System::PacketBuffer::Free(buffer);
 }
 
+void RendezvousServer::OnRendezvousStatusUpdate(Status status, CHIP_ERROR err)
+{
+    VerifyOrExit(err == CHIP_NO_ERROR, ChipLogError(AppServer, "OnRendezvousStatusUpdate: %s", chip::ErrorStr(err)));
+
+    switch (status)
+    {
+    case RendezvousSessionDelegate::SecurePairingSuccess:
+        ChipLogProgress(AppServer, "Device completed SPAKE2+ handshake");
+        SessionManager().NewPairing(Optional<Transport::PeerAddress>{}, &mRendezvousSession.GetPairingSession());
+        break;
+    case RendezvousSessionDelegate::NetworkProvisioningSuccess:
+        ChipLogProgress(AppServer, "Device was assigned network credentials");
+        break;
+    default:
+        break;
+    };
+
+exit:
+    return;
```

Comment:
I think we had some clang-tidy removals of redundant control flow

@mspang, any chance we can get that stuff into CI?

Reply:
It's not redundant. You need to have some instruction after a label so "return" serves as a no-op in such a case.

Reply:
ok, eesh

Reply:
I would welcome a SWTT discussion on when the VerifyOrExit pattern makes sense or not. I am unsure if there is a lot of difference between an inline return at line 127 vs a goto to a line that just does return.

Reply:
I'm not a big fan of VerifyOrExit etc. as they force me to declare variables at a scope beginning, but I admit they can make code more concise than a bunch of "if ... return"-s and they're widely used in CHIP, so I just conformed to the standard. But I'll carefully listen to the discussion and adjust :)